### DISCUSSION TITLE
Spreadsheets Insert Row [Help Wanted]

### DISCUSSION THREAD

**Discussion by the-questjon** [standardnotes/forum#2840] [General]
Unable to insert rows into spreadsheet

1. Click on the desired row
2. Select the insert tab
3. Click on either the 'insert row above' or 'insert row below' icon
4. See error: Cannot insert cells due to data loss possibility. Select another insert location or delete the data from the end of your worksheet.

**Expected behavior**

A row should be able to be inserted either above or below the selected row

**Screenshots**

See attached screenshot

**Desktop (please complete the following information):**

Affects all apps for Windows, Linux Flatpak, or web. Latest release versions.

![Screenshot (4)](https://user-images.githubusercontent.com/86742897/180796743-3801773c-0cce-49de-84d4-ad369312d690.png)

**Comment by adrihut** [comment]
Hi @thejonsalt! This error is triggered as a warning...

- When there's data at the last row/column of a sheet, and you try to insert a new row/column within the sheet; or
- When you try to insert a certain number of rows/columns that would push your current data beyond the edge of the sheet

In either case, you can still insert multiple rows using the "Add Row/Column" buttons. Just need to make sure that multiple consecutive rows/columns are selected before you click on the button.
ㅤ
Hope this helps!

**Comment by adamhotep** [comment]
I have the same issue. On the web app, I can't insert a row above or below a selected cell or row or consecutive rows, even from the 75th (and final) row, which is blank. I also cannot delete the final row. 

On the Android app, I get "Cannot insert row below the last row" and adding before doesn't do anything (the button darkens to indicate it has been pressed, but then nothing. It stays dark until I do something else, like trying to add below). Adding after the penultimate row is the same non-action.

Adding columns sometimes has the same error, but it's working right now. 

My iPhone has died, but as of a month or two ago, it had similar issues. IIRC, it couldn't create columns or rows anywhere, so I had to do that from the web app on my laptop.

Update: After fiddling with it on both web and Android, now the web app acts like the Android app (complaining about adding at the end, adding row above or below is a no-op), though I can still add columns from the web.

**Comment by adamhotep** [comment]
**I found a workaround!** (big enough update to make a second comment rather than revising the last one)

Add another sheet (e.g. Sheet2) to the spreadsheet note (use the <kbd>➕</kbd> button on the bottom left of the spreadsheet) and add rows there. At a certain point, you can only add one row at a time, but for some reason (a bug?) this adds rows to the other sheet.

**Comment by ajilityj** [comment]
why was this issue coverted to a discussion? 2 years later and it's still an issue on Android (app & web). the workaround is obnoxious.
